### PR TITLE
create seperate test contracts

### PR DESCRIPTION
**Pull request by Jim8y**
Closes https://github.com/neo-project/neo-devpack-dotnet/issues/866

### CODE REVIEW

**Comment by Jim8y** [comment]
@lock9

**Comment by lock9** [comment]
I initialized the neo submodule, but I still have some dependency issues. I'm unable to build the project:


Screenshot 2024-01-17 at 13 49 35


Any clues?

**Comment by cschuchardt88** [comment]
Did you run `git submodule update --remote` ?

**Comment by Jim8y** [comment]
@superboyiii  mind to take a look?

**Comment by superboyiii** [comment]
@Jim8y It works well on mine.

**Comment by Jim8y** [comment]
@shargon

File: tests/Neo.Compiler.CSharp.TestContracts/Neo.Compiler.CSharp.TestContracts.csproj [hunk 1]
```diff
@@ -0,0 +1,202 @@
+<Project Sdk="Microsoft.NET.Sdk">
+
+    <PropertyGroup>
+        <TargetFramework>net7.0</TargetFramework>
+        <ImplicitUsings>enable</ImplicitUsings>
+        <Nullable>enable</Nullable>
+    </PropertyGroup>
+
+    <ItemGroup>
+      <None Include="Contract1.cs">
```
Comment (shargon) [inline tests/Neo.Compiler.CSharp.TestContracts/Neo.Compiler.CSharp.TestContracts.csproj:10]: Can we use *.cs?

File: tests/Neo.SmartContract.Framework.TestContracts/Neo.SmartContract.Framework.TestContracts.csproj [hunk 2]
```diff
@@ -0,0 +1,85 @@
+<Project Sdk="Microsoft.NET.Sdk">
+
+    <PropertyGroup>
+        <TargetFramework>net7.0</TargetFramework>
+        <ImplicitUsings>enable</ImplicitUsings>
+        <Nullable>enable</Nullable>
+    </PropertyGroup>
+
+    <ItemGroup>
+      <None Include="Contract_Attribute.cs">
```
Comment (shargon) [inline tests/Neo.SmartContract.Framework.TestContracts/Neo.SmartContract.Framework.TestContracts.csproj:10]: Same here

File: tests/Neo.SmartContract.Framework.UnitTests/Utils/Extensions.cs [hunk 3]
```diff
@@ -5,6 +5,7 @@ namespace Neo.SmartContract.Framework.UnitTests.Utils
 {
     static class Extensions
     {
+        public static readonly string TestContractRoot = "../../../../Neo.SmartContract.Framework.TestContracts/";
         public static void ClearNotifications(this ApplicationEngine engine)
```
Comment (shargon) [inline tests/Neo.SmartContract.Framework.UnitTests/Utils/Extensions.cs:9]:
```suggestion
        public static readonly string TestContractRoot = "../../../../Neo.SmartContract.Framework.TestContracts/";

        public static void ClearNotifications(this ApplicationEngine engine)
```

File: tests/Neo.Compiler.CSharp.TestContracts/Neo.Compiler.CSharp.TestContracts.csproj (hunk 1 above)
Comment (Jim8y) [reply tests/Neo.Compiler.CSharp.TestContracts/Neo.Compiler.CSharp.TestContracts.csproj:10]: I did not pay attention to this, will update.

**Comment by Jim8y** [comment]
Puiiiii

Error: Bad response: 503 <h2>This website is under heavy load (queue full)</h2><p>We're sorry, too many people are accessing this website at the same time. We're working on this problem. Please try again later.</p>

File: tests/Neo.Compiler.CSharp.UnitTests/Neo.Compiler.CSharp.UnitTests.csproj [hunk 4]
```diff
@@ -9,13 +9,13 @@
...
-    <Compile Remove="TestClasses\*.cs" />
-    <None Include="TestClasses\*.cs">
-      <CopyToOutputDirectory>PreserveNewest</CopyToOutputDirectory>
-    </None>
-  </ItemGroup>
+<!--  <ItemGroup>-->
+<!--    <None Include="..\Neo.Compiler.CSharp.TestContracts\*.cs">-->
+<!--      <CopyToOutputDirectory>PreserveNewest</CopyToOutputDirectory>-->
+<!--    </None>-->
+<!--  </ItemGroup>-->
```
Comment (shargon) [inline tests/Neo.Compiler.CSharp.UnitTests/Neo.Compiler.CSharp.UnitTests.csproj:19]: Remove them?

File: tests/Neo.SmartContract.Framework.UnitTests/Neo.SmartContract.Framework.UnitTests.csproj [hunk 5]
```diff
@@ -9,13 +9,13 @@
...
-    <Compile Remove="TestClasses\*.cs" />
-    <None Include="TestClasses\*.cs">
-      <CopyToOutputDirectory>PreserveNewest</CopyToOutputDirectory>
-    </None>
-  </ItemGroup>
+<!--  <ItemGroup>-->
+<!--    <None Include="../Neo.SmartContract.Framework.TestContracts\*.cs">-->
+<!--      <CopyToOutputDirectory>PreserveNewest</CopyToOutputDirectory>-->
+<!--    </None>-->
+<!--  </ItemGroup>-->
```
Comment (shargon) [inline tests/Neo.SmartContract.Framework.UnitTests/Neo.SmartContract.Framework.UnitTests.csproj:19]: Remove them?

**Review by shargon — APPROVED** [review]

**Pull request merged**